packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -337,14 +380,37 @@ class SnackBar extends StatefulWidget {
       duration: duration,
       animation: newAnimation,
       onVisible: onVisible,
+      useScaffoldMessenger: useScaffoldMessenger,
+    );
+  }
+
+  /// Creates a copy of this snack bar but with useScaffoldMessenger set to true.
+  ///
+  /// If the original snack bar lacks a key, the newly created snack bar will
+  /// use the given fallback key.
+  SnackBar withScaffoldMessenger({ Key fallbackKey }) {
```

Comment:
This is just temporary? Should it be deprecated?

Reply:
Added.